src/Resource/CloudAccount/Backup.php

```diff
@@ -60,46 +60,23 @@ class Backup extends Model {
    * @throws CloudAccountException
    */
   public function delete() : void {
-    if (! $this->isReal()) {
-      throw new CloudAccountException(
-        CloudAccountException::INVALID_BACKUP,
-        ['action' => __METHOD__]
-      );
-    }
-
     $endpoint = $this->_getEndpoint();
     assert($endpoint instanceof Endpoint);
```

Comment:
Why are we using assert here?

Reply:
this is not new code. see https://github.com/nexcess/nexcess-cli/pull/15#issue-229081646

This is primarily for static analysis (to let phan know it is indeed the subclass of object we expect), but because assertions in php 7+ are zero-cost in production, there's no harm in leaving them in.